src/Controls/src/Core/Platform/Windows/Extensions/FormattedStringExtensions.cs

```diff
@@ -127,5 +130,56 @@ public static void UpdateInlines(this TextBlock textBlock, Label label)
 
 			return Tuple.Create(run, span.TextColor, span.BackgroundColor);
 		}
+
+		public static void RecalculateSpanPositions(this TextBlock control, Label element, IList<double> inlineHeights)
+		{
+			if (element?.FormattedText?.Spans == null
+				|| element.FormattedText.Spans.Count == 0)
+				return;
+
+			var labelWidth = control.ActualWidth;
+
+			if (labelWidth <= 0 || control.Height <= 0)
+				return;
+
+			for (int i = 0; i < element.FormattedText.Spans.Count; i++)
+			{
+				var span = element.FormattedText.Spans[i];
+
+				var inline = control.Inlines.ElementAt(i);
+
+				// TODO: Alternatives?. GetCharacterRect always return an empty Rect
```

Comment:
Here we have the pending key to complete the implementation. We need, as in other platforms, to calculate the bounding box of each Span to have the region that can detect the gesture.

Ideally, the GetCharacterRect method should be able to provide us with the Rect that borders a TextPointer. However, this always returns us an empty Rect.

Reply:
@PureWeen Could we get feedback from the WinUI Team?

Reply:
I wonder if this is only set once the label is on screen/loaded/template applied?